### PR TITLE
Task-88: Create Event page UX Styling Cleanup

### LINKED ISSUE #316: TASK-88: CreateEvent Page UX/Styling Cleanup
**Task Description**

As a developer, I want to fix the styling and improve the UX of the CreateEvent page:
- [ ] 87.1 Move “Location” above “Description”
- [ ] 87.2 Add “Restriction” title above pillbox
- [ ] 87.3 Pillbox should grow along with the input boxes
- [ ] 87.4 Slightly wider spacing between title and inputs
- [ ] 87.5 When done creating an event, redirect to My Events (not Home/Discover)
- [ ] 87.6 When image is not added to form, then user must see image error message

**Acceptance Criteria**
- [ ] Documentation present where necessary
- [ ] Unit and UI tests are implemented for every new component and function
- [ ] The application should still work with seeded data

### PR DESCRIPTION
**Pull request by ys-lin**
Closes #316 

How to test: please check if every acceptance criteria is satisfied.

### CODE REVIEW

File: apps/mull-ui/src/app/pages/create-event/create-event.scss [hunk 1]
```diff
@@ -92,13 +93,26 @@
...
+  margin-top: 1rem;
+  .pill-options {
+    width: 100%;
+    margin-top: 0.5rem;
+    .pill-option {
+      width: 100%;
+    }
+  }
+}
+
```
Comment (r-saba) [inline apps/mull-ui/src/app/pages/create-event/create-event.scss]: We need remove the left and right border on the second pill. This will do the trick.

```suggestion
.pill-options-container {
  width: 100%;
  margin-top: 1rem;
  .pill-options {
    width: 100%;
    margin-top: 0.5rem;
    button:nth-of-type(2) { // remove border around second pill
      border-right: 0;
      border-left: 0;
    }
    .pill-option {
      width: 100%;
    }
  }
}
```

**Review by r-saba — COMMENTED** [review]
In `create-event.tsx` can you change the `p` tag with the text "Create Event" to a `h1`.

File: apps/mull-ui/src/app/pages/create-event/create-event.tsx [hunk 2]
```diff
@@ -146,7 +150,7 @@ const CreateEventPage = ({ history }: CreateEventProps) => {
           },
         });
         updateToast('Event Created', toast.TYPE.SUCCESS);
-        history.push(ROUTES.HOME);
+        history.push(ROUTES.EVENT_BY_ID.replace(':id', `${eventId}`));
```
Comment (cristian-aldea) [inline apps/mull-ui/src/app/pages/create-event/create-event.tsx]: ⚠ Careful, the acceptance criteria is to redirect back to `/home/myevents`. However, this currently probably won't work. After you route to `/home/myevents` it seems it always redirects back to `/home/discover`. To fix this, you need to remove the following code from lines 104-106:
```tsx
                <PrivateRoute exact path={ROUTES.HOME}>
                  <Redirect to={ROUTES.DISCOVER.url} />
                </PrivateRoute>
```

Comment (ys-lin) [reply apps/mull-ui/src/app/pages/create-event/create-event.tsx]: oh this one is when you created a event, it redirects you to the event itself. Sorry  I misunderstood the specification, it says to redirect to my event, didn't know it's the my events page. I don't think the lines in create-event you highlighted has anything to do with the bug you pointed out

Comment (ys-lin) [reply apps/mull-ui/src/app/pages/create-event/create-event.tsx]: It's fixed now :3

**Review by r-saba — APPROVED** [review]
LGTM! Good work!

File: apps/mull-ui/src/app/pages/create-event/create-event.scss [hunk 3]
```diff
@@ -92,13 +93,29 @@
...
+    width: 100%;
+    margin-top: 0.5rem;
+    button:nth-of-type(2) { // remove border around second pill
+      border-right: 0;
+      border-left: 0;
+    }
+    .pill-option {
+      width: 100%;
+    }
+  }
```
Comment (cristian-aldea) [inline apps/mull-ui/src/app/pages/create-event/create-event.scss]: 💡 This styling should be integrated into `pill-options.scss`, since it's changing the way the insides of the component are styled.

In general, if you're changing the insides of the styling of a component, the component itself should be responsible, not its parent.

File: apps/mull-ui/src/app/app.tsx [hunk 4]
```diff
@@ -90,8 +90,8 @@ export const App = () => {
           />
           <Route exact path={ROUTES.TOKEN_REDIRECT} component={TokenRedirectPage} />
           <PrivateRoute exact path={ROUTES.CREATE_EVENT} component={CreateEventPage} />
-          <Route exact path={ROUTES.EVENT_BY_ID} children={<EventPage />} />
-          <PrivateRoute path={ROUTES.HOME}>
+          <Route exact path={ROUTES.EVENT_BY_ID} children={<EventPage prevPage="Discover" />} />
```
Comment (cristian-aldea) [inline apps/mull-ui/src/app/app.tsx]: 💡 I'd say you can revert this change, we can't know for sure that the discover page was the last page for the user

**Review by cristian-aldea — APPROVED** [review]
LGTM, everything is addressed. Had a few suggestions but nothing critical

File: apps/mull-ui/src/app/pages/create-event/create-event.scss (hunk 3 above)
Comment (ys-lin) [reply apps/mull-ui/src/app/pages/create-event/create-event.scss]: oh you're right, I will move it now
